spec/close_milestone_action_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::CloseMilestoneAction do
+  describe 'run' do
+    let(:test_repository) { 'test/test' }
+    let(:test_milestone_title) { 'milestone_title' }
+    let(:test_milestone) { { number: 1234 } }
+    let(:test_token) { 'GITHUB_TOKEN' }
+    let(:github_helper) { class_double(Fastlane::Helper::GithubHelper) }
+
+    before do
+      allow(github_helper).to receive(:update_milestone)
+      allow(github_helper).to receive(:get_milestone).and_return(test_milestone)
+      allow(Fastlane::Helper::GithubHelper).to receive(:new).and_return(github_helper)
+    end
+
+    it 'sets the github_token' do
+      expect(Fastlane::Helper::GithubHelper).to receive(:new).with(github_token: test_token)
+      described_class.run(mock_params)
+    end
+
+    it 'calls the get_milestone' do
+      expect(github_helper).to receive(:get_milestone).with(test_repository, test_milestone_title)
+      described_class.run(mock_params)
+    end
+
+    it 'calls the update_milestone' do
+      expect(github_helper).to receive(:update_milestone).with(
+        repository: test_repository,
+        number: 1234,
+        options: { state: 'closed' }
+      )
+
+      described_class.run(mock_params)
+    end
```

Comment:
That's a good test, but I wonder if the thing we want to test shouldn't be that we ultimately call the right GitHub API (well, in our case, the right `Octokit::Client` method).

The purpose of a unit test is to test ultimate behavior independently of the implementation details. So we want to test that a given call of our helper methods ends up calling the final endpoint, regardless of the internal implementation of the action, and if we use an intermediate helper class to achieve our goal of if the action ends up calling `Octokit::Client` directly without using a helper, or whatnot. The end goal is that we call the right REST API / `Octokit::Client` method with the right parameters, so that the expected end result and action on GitHub happens in the end.

Given this, I'd suggest to rewrite your unit tests (this one and the others that have the same approach so far) to test the ultimate behavior independently of the implementation detail of the action using a helper under the hood or not.